### PR TITLE
Fixed line-height for input and textarea + use correct font

### LINKED ISSUE #10532: input's height is affected by parent's line-height
Test case:

``` html
<div style="line-height: 100px;">
    <input placeholder="foobar">
</div>
```

Firefox:
firefox

Servo:
servo

### PR DESCRIPTION
**Pull request by perlun**
Fixes #10532.

Replaces #10553 since @Sendoushi had problems getting a proper rebase locally, so helping him out to get this merged.

(Credits to Sendoushi who did the actual changes; I'm just helping out to getting things into a mergeable state.)

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10809)

### CODE REVIEW

**Comment by jdm** [comment]
Could you squash these?

**Comment by perlun** [comment]
Sure. Done.

**Comment by Sendoushi** [comment]
Well thanks @perlun you've been of extreme help on this matter :) And sorry for the time delay for something so simple. Life crazy around here ehehe

**Comment by perlun** [comment]
> Well thanks @perlun you've been of extreme help on this matter :)

Glad I could help! And thanks a lot for the contribution.

> And sorry for the time delay for something so simple. Life crazy around here ehehe

No worries whatsoever.

**Comment by perlun** [comment]
@jdm - any idea how we can sort this mess out? I cannot really see that this PR changes stuff that affects these tests, but maybe I'm blind.

**Comment by jdm** [comment]
I downloaded http://build.servo.org/builders/mac-rel-css/builds/507/steps/test/logs/test-css.log/text (which is the result of appending /text to the URL for "test-css.log" on the buildbot page) and ran it through the [reftest analyzer](https://github.com/servo/servo/blob/master/tests/wpt/README.md#analyzing-reftest-results):
screen shot 2016-04-25 at 2 52 30 pm
screen shot 2016-04-25 at 2 52 45 pm

**Comment by jdm** [comment]
So that's fun. There's a tiny little extra bit of space on the left side of the third input in both testcases. This feels like it's tickling a bug in our layout code, unfortunately :(

**Comment by perlun** [comment]
Small but subtle differences... @Sendoushi, any suggestions?

**Comment by jdm** [comment]
I'm inclined to file an issue for the two failing tests, mark them as expected to fail, and merge these changes.

**Comment by Sendoushi** [comment]
@perlun I would need to have a check on the code. Sometimes what happens is that people forget that half-pixels don't work that well on displays. These kind of elements should always have an odd width but this is just a possibility without actually checking anything!

**Comment by perlun** [comment]
@jdm - I updated the test `.ini` files with the new expected failures like you suggested. Please r+ once Travis is done so we can get this ball rolling.

File: tests/wpt/metadata-css/css-transforms-1_dev/html/transform-input-018.htm.ini [hunk 1]
```diff
@@ -1,5 +1,3 @@
 [transform-input-018.htm]
   type: reftest
-  expected:
-    if os == "mac": PASS
-    FAIL
+  expected: FAIL
```
Comment (jdm) [inline tests/wpt/metadata-css/css-transforms-1_dev/html/transform-input-018.htm.ini]: Could you add `bug: https://github.com/servo/servo/issues/10881` to both of these manifests as well?

Comment (perlun) [inline tests/wpt/metadata-css/css-transforms-1_dev/html/transform-input-018.htm.ini]: Done.

**Comment by KiChjang** [comment]
@bors-servo retry

* The annoying git

**Comment by jdm** [comment]
@bors-servo: retry
* infra

**Comment by KiChjang** [comment]
@bors-servo retry

* The annoying git

**Comment by jdm** [comment]
Not sure why those failures wouldn't have appeared before, but the failing reftests are comparing input elements with type="button" against button elements. We should apply the CSS changes here to buttons as well to avoid these failures.

**Comment by perlun** [comment]
Blargh. Apparently the focus test still fails. Have to look into that tomorrow, bedtime now.

**Comment by perlun** [comment]
@jdm - any idea what could be causing that?

**Comment by jdm** [comment]
The button in submit_focus_a.html now has padding applied to it, and submit_focus_b.html is explicitly setting the padding for the reference element to 0. I think we should get rid of all the explicit styling in both submit_focus_a.html and submit_focus_b.html, since I think it's just there to compensate for the missing styles that are being added by this PR.

**Comment by perlun** [comment]
Awesome, thanks! Updated now, so let's hope that the builds are now all :green_heart: again.

**Comment by emilio** [comment]
Well, that's easy to test :). I'll let @jdm to re-check the changes, but meanwhile:

@bors-servo: try

**Pull request merged**